crates/latch_core/src/ecs/mod.rs

```diff
@@ -2,6 +2,70 @@
 //
 // Runtime-defined component system with TypeScript compatibility.
 // See docs/ecs-runtime-components.md for architectural details.
+//
```

Comment:
A large header comment like this is better off in the docs than inline. Individual classes or methods can and should have small block comment descriptions, but making a huge comment block like this just means more for people to scroll past.

We have a dedicated (though disorganized) `docs/` directory for usage details like this.